lxd/storage/drivers/drivers_lvm_volumes.go

```diff
@@ -0,0 +1,732 @@
+package drivers
+
+import (
+	"fmt"
+	"io"
+	"math"
+	"os"
+
+	"github.com/pkg/errors"
+	"golang.org/x/sys/unix"
+
+	"github.com/lxc/lxd/lxd/migration"
+	"github.com/lxc/lxd/lxd/operations"
+	"github.com/lxc/lxd/lxd/revert"
+	"github.com/lxc/lxd/lxd/rsync"
+	"github.com/lxc/lxd/shared"
+	log "github.com/lxc/lxd/shared/log15"
+)
+
+// GetVolumeUsage returns the disk space used by the volume (this is not currently supported).
+func (d *lvm) GetVolumeUsage(vol Volume) (int64, error) {
+	return 0, ErrNotSupported
+}
+
+// ValidateVolume validates the supplied volume config.
+func (d *lvm) ValidateVolume(vol Volume, removeUnknownKeys bool) error {
+	return d.validateVolume(vol, nil, removeUnknownKeys)
+}
+
+// HasVolume indicates whether a specific volume exists on the storage pool.
+func (d *lvm) HasVolume(vol Volume) bool {
+	volDevPath := d.lvmDevPath(d.config["lvm.vg_name"], vol.volType, vol.contentType, vol.name)
+	volExists, err := d.logicalVolumeExists(volDevPath)
+	if err != nil {
+		return false
+	}
+
+	return volExists
+}
+
+// GetVolumeDiskPath returns the location of a disk volume.
+func (d *lvm) GetVolumeDiskPath(vol Volume) (string, error) {
+	return "", ErrNotImplemented
+}
+
+// CreateVolume creates an empty volume and can optionally fill it by executing the supplied filler function.
+func (d *lvm) CreateVolume(vol Volume, filler *VolumeFiller, op *operations.Operation) error {
+	if vol.contentType != ContentTypeFS {
+		return fmt.Errorf("Content type not supported")
```

Comment:
I believe we've been just returning ErrNotSupported in the other drivers for such cases now.